examples/deepcopy-gen/generators/deepcopy.go

```diff
@@ -523,10 +523,10 @@ func (g *genDeepCopy) doMap(t *types.Type, sw *generator.SnippetWriter) {
 				sw.Do("}\n", nil)
 				sw.Do("(*out)[key] = *newVal\n", nil)
 			} else {
-				sw.Do("if newVal, err := c.DeepCopy(&val); err != nil {\n", nil)
-				sw.Do("return err\n", nil)
-				sw.Do("} else {\n", nil)
+				sw.Do("if newVal, err := c.DeepCopy(&val); err == nil {\n", nil)
```

Comment:
It's much more idiomatic to test err != nil - why change that?  I had a hard time realizing what you changed because my eyes just skip the pattern.  I'd rather see a test for != nil and a `return nil` in the success path